### PR TITLE
Implement AWS_BYTE_CUR_INIT_FROM_LITERAL

### PR DESCRIPTION
**Pull request by ColdenCullen**
This is mostly for unit testing stuff.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ColdenCullen** [comment]
Hang on this isn't as useful as I thought it would be

**Comment by ColdenCullen** [comment]
Changed my mind, rather than defining a static, it's more useful to have an init macro.

File: include/aws/common/byte_buf.h [hunk 1]
```diff
@@ -440,6 +440,11 @@ const uint8_t *aws_lookup_table_to_lower_get(void);
 
 AWS_EXTERN_C_END
 
+/**
+ */
+#define AWS_BYTE_CUR_INIT_FROM_LITERAL(literal)                                                                  \
+    { .ptr = (uint8_t *)(literal), .len = AWS_ARRAY_SIZE(literal) }
```
Comment (graebm) [inline include/aws/common/byte_buf.h]: won't AWS_ARRAY_SIZE of "a" be 2? because of the '\0'?

Comment (ColdenCullen) [reply include/aws/common/byte_buf.h]: Yes! That's probably not what we want here huh.

**Pull request merged**